doc/api/esm.md

```diff
@@ -120,12 +129,15 @@ given module specifier and parent file URL:
 const baseURL = new URL('file://');
 baseURL.pathname = process.cwd() + '/';
 
-export async function resolve(specifier,
-                              parentModuleURL = baseURL,
-                              defaultResolver) {
+export default function (parent) {
   return {
-    url: new URL(specifier, parentModuleURL).href,
-    format: 'esm'
+    async resolve(specifier,
```

Comment:
its called resolve because it resolves the location of a module request. async so you can do whatever magic you need to do e.g. reading files (package.json for instance) without blocking

Reply:
ok - so it doesn't have to be an async function but its return value will always be `await`ed?

Reply:
actually that brings up a good point. in the loader we do `await` but people making hooks might not know to use the same abstraction when calling parent hooks

Reply:
well it is always async, you do need to await it if you aren't just passing it through (ala `return parent.resolve(...)`).